core/macros/src/lib.rs

```diff
@@ -41,17 +41,21 @@ pub fn enum_trait_object(args: TokenStream, item: TokenStream) -> TokenStream {
     let enum_input = parse_macro_input!(args as ItemEnum);
     let enum_name = enum_input.ident.clone();
 
-    if trait_generics.lifetimes().count() > 1 {
-        panic!("Only one lifetime parameter is currently supported");
-    }
+    assert!(
+        trait_generics.lifetimes().count() <= 1,
+        "Only one lifetime parameter is currently supported"
+    );
+
+    assert!(
+        trait_generics.type_params().count() <= 1, // XXX shouldn't this be <= 0; or rather, == 0; or similar to .is_empty() or .has_next()?
```

Comment:
Though @Herschel would know the best, from my understanding the first two asserts are not needed at all, because type generics (lifetimes + type params) are seemed to be handled properly down below.
If we would like to be on the safe-side, I suggest turning the `// XXX` comment into a clear `TODO` that questions about these two, mentioning the discrepancy between the second assert condition and its message.